drivers/gpu/drm/vc4/vc4_vec.c

```diff
@@ -372,11 +409,19 @@ static int vc4_vec_connector_get_modes(struct drm_connector *connector)
 	return 1;
 }
 
+static void vc4_vec_connector_reset(struct drm_connector *connector)
+{
+	drm_atomic_helper_connector_reset(connector);
+	/* preserve TV standard */
+	if (connector->state)
+		connector->state->tv.mode = vc4_vec_get_default_mode(connector);
```

Comment:
Do we want to reset to the default (as defined by the module parameter), or retain the old setting?
If you change the module parameter (it is permissions 0600) then on the next reset call don't we have the same scenario as at boot where the mode may not match the standard?

Reply:
You're right about the comment - checkpatch says "WARNING: Block comments use a trailing */ on a separate line".

Reply:
reset is only called once at boot time, it's what creates the first initial commit. Any state after that is the union of the previous state and whatever has been changed in the various commits that occurred.

Reply:
Uh, this is actually one of the places where I don't exactly know what I'm doing. From what I understand, this function is called when a new connector "state" object is created. By default, it is more or less just zeroed out, which means also setting `tv.mode` field to 0, which is NTSC in our case. The default value set through `drm_object_attach_property()` is not respected for some reason, so without this code, we would end up in NTSC mode at boot regardless of what was set there.

Setting this value later in userspace (`xrandr --set mode PAL` etc.) still works.

Actually the only other driver in the entire kernel codebase that uses the `tv.mode` field is i915 - which defaults to NTSC anyway, as pointed out in https://github.com/raspberrypi/linux/pull/4241#pullrequestreview-691627291. nouveau and ch7006 just keep track of it in a separate private variable and use that instead.

I don't know, maybe a cleaner solution would be to patch the DRM subsystem itself so that it would pull the default value from whatever has been set through `drm_object_attach_property()` - because not respecting that even though the `tv_mode_property` is automatically mapped to that field seems like a bug. Maybe that should be done directly in `vc4_vec_connector_reset()` or something, although I'm not sure if that's even possible.